test/Microsoft.AspNetCore.Mvc.Abstractions.Test/ModelBinding/ModelStateDictionaryTest.cs

```diff
@@ -510,8 +599,8 @@ public void GetFieldValidity_ReturnsInvalid_IfAllKeysAreValidatedAndAnyEntryIsIn
         {
             // Arrange
             var dictionary = new ModelStateDictionary();
-            dictionary["user.Address"] = new ModelStateEntry { ValidationState = ModelValidationState.Valid };
-            dictionary["user.Name"] = new ModelStateEntry { ValidationState = ModelValidationState.Valid };
+            dictionary.MarkFieldValid("user.Address");
+            dictionary.MarkFieldValid("user.Name");
```

Comment:
This is why I wasn't really concerned about the pivot on `IDictionary`. These convenience methods are far more useful than the raw api.